Model/Indexer/Category.php

```diff
@@ -82,6 +82,7 @@ public function execute($categoryIds)
                 );
             } else {
                 $this->queue->addToQueue($this->fullAction, 'saveConfigurationToAlgolia', ['store_id' => $storeId], 1);
+                $categoryIds = [];
```

Comment:
Adding condition on line 66 like:
```php
if ($categoryIds === null) {
    $categoryIds = [];
}
```
and changing the `if ($categoryIds !== null)` conditon would be much more readable. What do you think, @mwr?

Reply:
Do you mean before the `if ($categoryIds !== null) ` in line 75?

If it is before the if in line 75 it would change the behaviour since the deleteObjects queue entry would also be added. 
With the solution above if categoryIds === null the deleteObjects will not be queued.

After the if, I see no problem in adding it there.

Reply:
I know, my idea was to change the condition as well to still work consistently.

But I'm wondering - the `null` is issues only for line 92: `count($categoryIds)`. How about just changing this line to:

```php
is_array($categoryIds) ? count($categoryIds) : 1
```

I think that'd be better for readability as well.

Reply:
ok, now I understand what you mean.
Sure that is definitely easier to read and understand.

Reply:
Cool! I'll wait for your commit then :) Or let me know if you want me to do it :)

Reply:
Feel free to do it. I won't get to this until thursday